File: ch05/DatePick/DatePick/ViewController.m
//
//  ViewController.m
//  DatePick
//
//  Created by MyZiyue on 2018/12/13.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIDatePicker *datePicker;
@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (IBAction)onClick:(id)sender {
    NSDate* theDate = self.datePicker.date;
    NSLog(@"The date picked is : %@", [theDate descriptionWithLocale:[NSLocale currentLocale]]);
    
    NSDateFormatter* dateFormatter = [[NSDateFormatter alloc] init];
    [dateFormatter setDateFormat:@"YYYY-MM-dd HH:mm:ss"];
    NSLog(@"The date fromate is : %@", [dateFormatter stringFromDate:theDate]);
    
    self.label.text = [dateFormatter stringFromDate:theDate];
}

@end

File: ch05/SDatePick/SDatePick/ViewController.swift
//
//  ViewController.swift
//  SDatePick
//
//  Created by MyZiyue on 2018/12/13.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var datePicker: UIDatePicker!
    @IBOutlet weak var label: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }


    @IBAction func onClick(_ sender: Any) {
        let theDate: Date = self.datePicker.date as Date
        let desc = theDate.description(with: NSLocale.current)
        NSLog("The date picked is : %@", desc)
        
        let dateFormatter: DateFormatter = DateFormatter()
        dateFormatter.dateFormat = "YYYY-MM-dd HH:mm:ss"
        NSLog("The date formate is : %@", dateFormatter.string(from: theDate as Date))
        
        self.label.text = dateFormatter.string(from: theDate as Date)
    }
}

